### PR TITLE
fix(pfd): FMA Showing FDs Engaged While ADIRs Not Aligned

### LINKED ISSUE #4242: Fully Simulated Automatic Flight Director Engagement after Power-On
**Is your feature request related to a problem? Please describe.**
No. Just a little accuracy check based on actual aircraft behavior

**Describe the feature you'd like to see implemented**
Engage the Flight Directors automatically after power-up

with delay of around 12 seconds according to the youtube video

**References**

A320 NEO AMM 22-10-00-00 AP/FD ENGAGEMENT - Description and Operation
Capture

Airbus A320 Standards Document
unknown

https://youtu.be/Mo7A3CuVb8g?t=48

**Additional context**

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #4242 (actually Andreas already did most of it)

## Summary of Changes

FDs are automatically armed by the FMGC on power-up. The FDs should not be engaged on the ground until at least 2 ADIRUs are aligned. @davidwalschots may want to pull the ADIRs change into his tree for #5242 and sequence this PR after it. Noticed by Ryan_D1994.

## Screenshots (if necessary)

Before:
image

After:
image

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by davidwalschots** [comment]
I think the dependency here is reversed. You should wait for #5242 to be merged, as `ADIRS_NUM_ALIGNED` is no longer needed once that PR is merged. Secondly, you could consider fixing this in cFPM, as it does indeed belong there and would remove the need to fix it multiple times.

File: src/systems/systems/src/adirs/mod.rs
Comment (davidwalschots) [inline src/systems/systems/src/adirs/mod.rs:254]: **thought**: Can make use of `filter` and `count`.
**suggestion**: See below, note that you need to add a `Write` implementation for `usize` for this in `systems/src/simulation/mod.rs`.

```suggestion
    fn number_aligned(&self) -> usize {
        self.adirus
            .iter()
            .filter(|adiru| adiru.is_aligned()).count()
    }
```

File: src/systems/systems/src/adirs/mod.rs
Comment (davidwalschots) [inline src/systems/systems/src/adirs/mod.rs:168]: **issue**: Missing `a320-simvars.md` documentation.

File: src/systems/systems/src/adirs/mod.rs
Comment (davidwalschots) [inline src/systems/systems/src/adirs/mod.rs:281]: **issue**: Missing unit test(s).

**Review by davidwalschots — CHANGES_REQUESTED** [review]

**Comment by 2hwk** [comment]
This PR has not had much activity recently/is stale, so to help us categorize I'm converting this back to draft. You can mark it as ready for review once there is a status update or change pushed.

**Comment by tracernz** [comment]
> This PR has not had much activity recently/is stale, so to help us categorize I'm converting this back to draft. You can mark it as ready for review once there is a status update or change pushed.

Unfortunately I think this really needs to be done in simulink, to prevent modes arming as well. I’ll keep it for now…

**Pull request closed without merging**
